src/NHibernate/Dialect/MsSqlCeDialect.cs

```diff
@@ -339,6 +339,9 @@ public override long TimestampResolutionInTicks
 		/// </summary>
 		public override bool SupportsPoolingParameter => false;
 
+		/// <inheritdoc/>
+		public override bool SupportsScalarSubSelects => false;
```

Comment:
Ms SQL CE supports only sub-queries defining a set, like with in `where exists(sub-query)`, `join (sub-query)`, `from (sub-query)`, `in (sub-query)`. Sub-queries to be handled as a scalar value (single column & single row) are not supported. So things like `select (sub-query) as colName`, `where (sub-query) = someValue`, `order by (sub-query)`, ..., are not supported.